app/controllers/application_controller.rb

```diff
@@ -343,8 +343,9 @@ def generic_exception(exception)
     if exception.try(:cause).is_a?(ActiveRecord::SubclassNotFound)
       sti_clean_up(exception.cause)
     else
-      Foreman::Logging.exception("Action failed", exception)
-      render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception => exception}
+      ex_message = exception.message
+      Foreman::Logging.exception(ex_message, exception)
+      render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception_message => ex_message, request_id: request.request_id.split('-').first}
```

Comment:
We must present both full UUID and partial UUID (first 8 characters) which we write to files by default. When using ElasticSearch, UUID must be searched as a whole string because it's fulltext. I am not sure if just 8 characters version would even work.